### ISSUE TITLE
Runtime-parameter value field does not re-adjust properly to GUI size (SF #199)

### ISSUE THREAD

**Issue by johann-petrak** [GateNLP/gate-core#25]
Bug created on SF on 2015-06-06

Still unsolved with GATE 8.4.1 and GATE 8.5 on Ubuntu 17.10 as of 2018-03-10.

## Original bug:

Seen this on Linux (Mate desktop), Gate 8.2. 
* show the runtime parameters of the Reset PR: The buttons for editing the list-valued values are shown on the right of the parameter table rows
* make the gui smaller: the buttons disappear though there is no space used up to their left
* make the gui larger than it was originally: the buttons get moved to the right too
* reduce the size of the GUI the the original size: the buttons do not move back to where they originally were and are now stuck even more to the right.

The bottom line is that the buttons never seem to move to the left, or that the value fields never adjust to become smaller, only larger. 
I think this is also a problem even when there are no buttons to the right: the value fields increase their size but do not decrease their size, even if empty.

**Comment by johann-petrak** [comment]
Also, why do those buttons have to be on the right side of the field anyways? If they were on the left, one could always use them without scrolling even if the fields extend beyond the visible area for some reason ..

**Comment by ianroberts** [comment]
I've made a small improvement to this now in bcd227c18935efc09a6b7469ebc32c601623c641 - the field still doesn't shrink when you narrow the window again, but at least now when you switch to a different PR in the application that resets the width to fit the current window size rather than being forever more stuck at the maximum width you ever reached.  Similarly, when creating a new PR instance the table grows when you widen the "new PR" window, doesn't shrink again when you narrow it, but at least resets when you close and re-open the window again.

I'm not really sure what the correct behaviour here should be - we can make the fields always track the parent width but everything I've tried for that means that they initially start far too narrow in the "new PR" window, and also you don't get the scroll bar for very long values (e.g. a really long `creole://` URI - the only way to view the part that has run off the right is to click in the field and then move the cursor to the right with the keyboard).

I can think of two sensible-ish alternatives to the current UI

 1. move the buttons into their own column in the table, to the left of the current text field column, then there's a trick you can use on an N-column JTable to make the left M columns fixed and the horizontal scroll bar only scroll the right N-M ones
 2. disable horizontal scrolling altogether but do the kind of thing that Excel does when you are editing particularly long formulas and have a "..." button to pop up a multi-line editor if the single line one is too cramped
